lenskit-core/specs/data-manifest.md

```diff
@@ -0,0 +1,111 @@
+# Data Manifest Schema
+
+This document describes the format of data manifests.
+
+Data manifests are JSON-style data.  Currently they are generally described in YAML files, but they may be described in other formats such as JSON.
+
+## Overall Layout
+
+A manifest file represents either a list or a map.
+
+If it is a map and contains at least one of the keys `file` or `type`, then the map is taken to describe a single data file (see below).
+
+Otherwise, the file describes a collection of data files.  If it is a map, then those files are labelled with their map keys; otherwise, they are labeled with their positions (starting from 0).
+
+For example:
+
+```yaml
+ratings:
+	file: "ratings.csv"
+	format: csv
+	entity: rating
+	header: true
+items:
+	file: "items.csv"
+	entity: movie
+	header: true
+	columns:
+	    movieId: id
+	    title: name
+```
+
+defines a data source consisting of ratings data, read from `ratings.csv`, and movie titles read from `items.csv`.  The files have labels.  The same manifest can be written with numeric labels as follows:
+
+```yaml
+-	file: "ratings.csv"
+	format: csv
+	entity: rating
+	header: true
+-	file: "items.csv"
+	entity: movie
+	header: true
+	columns:
+	    movieId: id
+	    title: name
+```
+
+## Data Source Description
+
+Individual data sources are described with the following schema.
+
+`type`
+:   The data source type.  Currently only `textfile` is supported.
+
+The remainder of the keys are defined by the particular data source.
+
+### `textfile` data sources
+
+`textfile` sources read data from text files, usually line-delimited.
```

Comment:
not clear what ussually line-delimited means here. In particular I think this can only support simple delineated files right? Like I don't even know if we support the more exotic flavors of csv (strings enclosing fields with commas and new lines in them, the commas and newlines should be ignored)